server.ts

```diff
@@ -70,49 +209,75 @@ function genTitle(
   return title;
 }
 function genMainContent(
-  post: AtoprotoAPI.AppBskyFeedDefs.PostView,
+  feed: AtoprotoAPI.AppBskyFeedDefs.FeedViewPost,
   usePsky: boolean,
   includeRepost: boolean,
+  fullMedia: boolean,
+  replyContext: boolean,
 ) {
+  const post = getPost(feed.post, fullMedia);
+  const reply = (AppBskyFeedDefs.isPostView(feed.reply?.parent))
+    ? getPost(feed.reply.parent, fullMedia)
+    : undefined;
+
   if (usePsky) {
     if (
       includeRepost &&
-      AppBskyEmbedRecord.isView(post.embed) &&
-      AppBskyEmbedRecord.isViewRecord(post.embed.record)
-    ) {
-      return ["[quote] ", uriToPostLink(post.embed.record.uri, usePsky)];
-    } else if (
-      AppBskyEmbedRecordWithMedia.isView(post.embed) &&
-      AppBskyEmbedRecord.isViewRecord(post.embed.record.record)
+      post.quote
     ) {
-      return ["[quote] ", uriToPostLink(post.embed.record.record.uri, usePsky)];
+      return ["[quote] ", uriToPostLink(post.quote.uri, usePsky)];
     }
 
     return [];
   }
 
-  if (!AppBskyFeedPost.isRecord(post.record)) {
-    return [];
-  }
-
-  const embedImages = getEmbedImages(post);
-  const imagesDiv = embedImages.length
-    ? tag("div", ...embedImages.map((image) => `<img src="${image.thumb}"/>`))
-    : "";
-
   return [
     "<![CDATA[",
-    imagesDiv,
-    tag("p", sanitize(post.record.text).replace(/\n/g, "<br>")),
-    (
-        AppBskyEmbedRecord.isView(post.embed) &&
-        AppBskyEmbedRecord.isViewRecord(post.embed.record) &&
-        AppBskyFeedPost.isRecord(post.embed.record.value)
+    post.media,
+    tag(
+      "p",
+      `<b>${sanitize(post.author.displayName || "")}</b> <i>@${
+        post.author.handle || "unknown"
+      }</i> <a href="${uriToPostLink(post.uri, usePsky)}">${
+        (post.isReply) ? "replied" : "posted"
+      }</a>:<br>`,
+      post.text,
+    ),
+    (post.quote)
+      ? tag(
+        "p",
+        `<br>[quote]<br><b>${
+          sanitize(post.quote.author.displayName || "")
+        }</b> <i>@${post.quote.author.handle || "unknown"}</i> <a href="${
+          uriToPostLink(post.quote.uri, usePsky)
+        }">${(post.quote.isReply) ? "replied" : "posted"}</a>:<br>`,
+        post.quote.media,
+        tag("p", post.quote.text),
       )
+      : "",
+    (replyContext && reply)
       ? tag(
         "p",
-        "<br>[quote]<br>",
-        sanitize(post.embed.record.value.text || "unknown"),
+        "<hr><hr>",
```

Comment:
It's a design issue, but multiple horizontal lines don't always display beautifully. Can you make one?
```suggestion
        "<hr>",
```

Reply:
I used double `<hr>`s to clearly show a separation of posts but I wasn't aware that some readers don't always display them beautifully. One `<hr>` will work fine in this generator.